2020/06/02/why-upgrading-software-libraries-is-imperative.html.md

```diff
@@ -0,0 +1,44 @@
+
+The application stack in question contained decade-old software and included extensive libraries when we received it from our client. The application consisted of languages including Java, Scala, Kotlin and JavaScript in tandem with an extensive list of libraries. The initial plan was to upgrade the complete stack all at once due to the gap between versions. This proved to be more difficult than expected due to a host of deprecated and removed functionalities in addition to their inter-dependency on a few of the libraries. 
+
+### Conflict Approach
+Speaking on upgrading libraries and dependencies,it is often said: “don't update unless you have to.” For the short term, this can be sustainable, however, not as much in the long run. Primarily, the updates protect from known vulnerabilities and cyber attacks. Scenarios arise where particular library fixes are implemented on the latest version, yet require upgrading other libraries to the latest version in a chain. Because an upgraded list of libraries needs complete testing and preparation for new issues, this directly impacts whether the application attempts to resolve an issue. Therefore, smaller and more frequent updates are more sustainable in the long run. Larger and more infrequent upgrades will not only result in unexpected errors, but also require regress testing to deliver a version free of error. The following points justify the necessity of software and library upgrades:
+
+* Apply timely security patches to reduce vulnerabilities and defend from cyberattack
+* Identify deprecated functionality earlier and use alternatives
+* Remain on your Long Term Support (LTS) version
+* Apply fixes on known bugs in the library
+
+### Benefits
+New versions of software libraries are periodically released which means the latest release superior to the former. Why don’t we make use of the better version when it is available. It is recommended to keep those softwares and libraries updated to get benefits such as bug fixes, new features, boosted performance, improvements on stability, compatibility, and security measures. Most of the time software/libraries upgraded are ignored in projects due to very minimal effects in appearance. However frequent updates deliver benefits which clearly demonstrate the importance of upgrading software libraries.
+
+### Real Time Difficulties
+In real-life scenarios, bug fixes and performance optimization of the application become a higher priority than upgrading libraries. In many cases, this results in a time discrepancy between the library’s version in application and the latest version upgrade. This discrepancy can lead to the following issues:
+
+* Unexpected error handling in new releases
+* Unexpected unsupported library dependency error
+* Complete end-to-end automated testing on library upgrade
```

Comment:
I don't understand this item -- how does having an old version affect any end-to-end automated testing?

Reply:
Sorry, It was misinterpreted on review. Updated the point as "Need to do complete end-to-end testing due to major version of library upgrade".